sycl/source/detail/context_impl.cpp

```diff
@@ -35,10 +35,8 @@ context_impl::context_impl(const vector_class<cl::sycl::device> Devices,
     DeviceIds.push_back(getSyclObjImpl(D)->getHandleRef());
   }
 
-  RT::PiResult Err;
-  PI_CALL((m_Context =
-      RT::piContextCreate(0, DeviceIds.size(), DeviceIds.data(), 0, 0, &Err),
-      Err));
+  PI_CALL(
+      RT::piContextCreate(0, DeviceIds.size(), DeviceIds.data(), 0, 0, &m_Context));
```

Comment:
Could there be a reason to check the result here?
If there is a rationale for not doing so, does it apply to several functions below as well?

Reply:
After the change the Err is still being checked in PI_CALL. It is just that we do not need a variable to store it anymore.
It is working the same way now as the other calls to apis in PI_CALL(api) which have pi_result as a return type.